src/core_plugins/region_map/public/choropleth_layer.js

```diff
@@ -181,85 +312,91 @@ export default class ChoroplethLayer extends KibanaMapLayer {
+      return {
+        leafletStyleFunction: () => {
+          return emptyStyle();
+        },
+        getMismatches: () => {
+          return [];
+        },
+        getLeafletBounds: () => {
+          return null;
+        }
+      };
+    }
 
+    const { min, max } = getMinMax(this._metrics);
 
-  if (data.length === 0) {
+    const boundsOfAllFeatures = new L.LatLngBounds();
     return {
-      getLeafletStyleFunction: function () {
-        return emptyStyle();
+      leafletStyleFunction: (geojsonFeature) => {
+        const match = geojsonFeature.__kbnJoinedMetric;
+        if (!match) {
+          return emptyStyle();
+        }
+        const boundsOfFeature = L.geoJson(geojsonFeature).getBounds();
+        boundsOfAllFeatures.extend(boundsOfFeature);
+
+        return {
+          fillColor: getChoroplethColor(match.value, min, max, this._colorRamp),
+          weight: this._lineWeight,
+          opacity: 1,
+          color: 'white',
+          fillOpacity: 0.7
+        };
       },
-      getMismatches: function () {
-        return [];
+      /**
+       * should not be called until getLeafletStyleFunction has been called
+       * @return {Array}
+       */
+      getMismatches: () => {
+        const mismatches = this._metrics.slice();
+        this._sortedFeatures.forEach((feature) => {
+          const index = mismatches.indexOf(feature.__kbnJoinedMetric);
+          if (index >= 0) {
+            mismatches.splice(index, 1);
+          }
+        });
+        return mismatches.map(b => b.term);
       },
       getLeafletBounds: function () {
-        return null;
+        return boundsOfAllFeatures.isValid() ? boundsOfAllFeatures : null;
       }
     };
-  }
-
-  const { min, max } = getMinMax(data);
-  const outstandingFeatures = data.slice();
-
-  const boundsOfAllFeatures = new L.LatLngBounds();
-  return {
-    getLeafletStyleFunction: function (geojsonFeature) {
-      let lastIndex = -1;
-      const match = outstandingFeatures.find((bucket, index) => {
-        lastIndex = index;
-        return bucket.term === geojsonFeature.properties[joinField];
-      });
 
-      if (!match) {
-        return emptyStyle();
-      }
-
-      outstandingFeatures.splice(lastIndex, 1);
+  }
 
-      const boundsOfFeature = L.geoJson(geojsonFeature).getBounds();
-      boundsOfAllFeatures.extend(boundsOfFeature);
+}
 
-      return {
-        fillColor: getChoroplethColor(match.value, min, max, colorramp),
-        weight: 2,
-        opacity: 1,
-        color: 'white',
-        fillOpacity: 0.7
-      };
-    },
-    /**
-     * should not be called until getLeafletStyleFunction has been called
-     * @return {Array}
-     */
-    getMismatches: function () {
-      return outstandingFeatures.map((bucket) => bucket.term);
-    },
-    getLeafletBounds: function () {
-      return boundsOfAllFeatures.isValid() ? boundsOfAllFeatures : null;
-    }
-  };
+function makeColorDarker(color) {
```

Comment:
this diff got really bad, most of this functions did not really change just their position in the file did ....